src/system.h

```diff
@@ -127,3 +127,7 @@
 #endif
 
 #endif
+
+#if defined(__APPLE__) && defined(__MACH__)
+#include <TargetConditionals.h>
```

Comment:
This adds the `TARGET_OS_OSX` and friends. On Macs there's this standard thing called "precompiled headers", which is essentially a header file that every compilation unit silently includes, and it's in here that this TargetConditionals is usually included. We don't have one of those, so this is as close as I could get.

Not strictly necessary, but I thought it was a bit cleaner than littering the code with these 3 lines every time we need to check if we're a Mac or not.